### ISSUE TITLE
merge doesn't account for many-to-many relationships

### ISSUE THREAD

**Issue by smcoll** [saxix/django-adminactions#108]
If the merged (deleted) item is related to via a ManyToManyField, those relationships are lost; only those related to the original are retained.  Hopefully, i will be providing more info soon- just wanted to get this logged.  Gotta do some data recovery first.

**Comment by smcoll** [comment]
Followup, to demonstrate.

```
# models.py
from django.db import models

class Foo(models.Model):
    name=models.CharField(max_length=100)

    def __unicode__(self):
        return self.name

class Bar(models.Model):
    name=models.CharField(max_length=100)
    foos=models.ManyToManyField(Foo, blank=True)

    def __unicode__(self):
        return self.name
```

Create some records:

```
>>> f1 = Foo.objects.create(name='f1')
>>> f2 = Foo.objects.create(name='f2')
>>> b1 = Bar.objects.create(name='b1')
>>> b1.foos.add(f1)
>>> b2 = Bar.objects.create(name='b2')
>>> b2.foos.add(f2)
>>> 
```

Now, i merge both `Bar` instances in the admin using the "dependencies: move" option.

```
>>> from foo.models import *
>>> b1 = Bar.objects.get(name='b1')
>>> b1.foos.all()
[<Foo: f2>]
>>> 
```

i was expecting to find `[<Foo: f2>, <Foo: f1>]`

Similarly, if i merge `Foo` objects.  First; resetting the data:

```
>>> Foo.objects.all().delete()
>>> Bar.objects.all().delete()
>>> f1 = Foo.objects.create(name='f1')
>>> f2 = Foo.objects.create(name='f2')
>>> b1 = Bar.objects.create(name='b1')
>>> b1.foos.add(f1)
>>> b2 = Bar.objects.create(name='b2')
>>> b2.foos.add(f2)
>>> 
```

Now i merge both `Foo` instances in the admin, and try out the related manager again:

```
>>> b1.foos.all()
[]
>>> b2.foos.all()
[<Foo: f1>]
```

i was expecting `[<Foo: f1>]` for both.

**Comment by int-ua** [comment]
Didn't verify the code above but verified that `_meta.many_to_many` https://github.com/saxix/django-adminactions/blob/develop/src/adminactions/api.py#L84 doesn't return reverse relations in my case too. AFAIU both `_meta.many_to_many` and `_meta.get_all_related_objects` should be [combined and] replaced with `_meta.get_fields` or, better (?) with `_meta._get_fields(forward=False, reverse=True)`.

**Comment by int-ua** [comment]
@saxix I can **try** doing it on the weekend if you want. But I cannot guarantee any Django <= 1.7 compatibility.

**Comment by int-ua** [comment]
Wait, is it already fixed in 1.4?

**Comment by int-ua** [comment]
Ok, I started already because I need it urgently. I've found that `_meta.many_to_many` is not intended for usage, they recommend `get_fields`:

https://docs.djangoproject.com/en/1.8/_modules/django/db/models/options/
```
    @cached_property
    def many_to_many(self):
        """
        Returns a list of all many to many fields on the model and its parents.

        Private API intended only to be used by Django itself; get_fields()
        combined with filtering of field properties is the public API for
        obtaining this list.
        """
        return make_immutable_fields_list(
            "many_to_many",
            (f for f in self._get_fields(reverse=False)
            if f.is_relation and f.many_to_many)
        )
```

'reverse=False' above means that it's intended only for relations defined on this model.
